packages/perspective-jupyterlab/src/ts/renderer.ts

```diff
@@ -47,15 +53,26 @@ const baddialog = () => {
 
 const WORKER = perspective.worker();
 export class PerspectiveDocumentWidget extends DocumentWidget {
-    constructor(options, type = "csv") {
+    _monitor: ActivityMonitor<DocumentRegistry.IModel, void>;
+    _psp: PerspectiveWidget & {
+        theme?: string;
+    };
+    _type: string;
```

Comment:
type is one of "csv", "arrow", or "json" so can be constrained to string literal orring.